### ISSUE TITLE
Remove OCM module dependencies from DNSPolicy

### ISSUE THREAD

**Issue by mikenairn** [Kuadrant/multicluster-gateway-controller#622]
Follow on from https://github.com/Kuadrant/multicluster-gateway-controller/issues/311

**What**

Remove the ocm module dependencies from the dnspolicy reconcilers: [dnspolicy_controller.go](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/controllers/dnspolicy/dnspolicy_controller.go#L26)  and [dnspolicy_dnsrecords.go](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go#L8).

**Why**

* Allow the policy to work independently of OCM or any other control plane. 
* Allow the policy to be moved to another repo without OCM being required.
* Ensure all OCM integration is performed by the gateway controller

**How**

Update the Gateway Controller to watch mangedclusterresources and add annotations to placed gateways on those clusters with any customisation such as geo or weighted values. Gateway controller should add the following annotation
```kuadrant.io/{managedcluster}/geo: "US"``` 
The policy controller need to be updated to look for these annotations on the gateway resources and use them for the associated DNS records. 
It should also be able to handle an annotation that has no {managedcluster} part as a single cluster value

**Comment by makslion** [comment]
for the last item - there is a follow up after #311 - the #496

**Comment by mikenairn** [comment]
> for the last item - there is a follow up after #311 - the #496 it could have a bit of an overlap

Remove last item in favour of https://github.com/Kuadrant/multicluster-gateway-controller/issues/496. This issue is now only concerned about making the "cluster" meta resource configurable for use when OCM is not the control plane of choice, or no control plane exists. Low priority at this moment in time.
